### DISCUSSION TITLE
Pre-splitting mesh not working with LowerDBlockFromSidesetGenerator

### DISCUSSION THREAD

**Discussion by ln53** [idaholab/moose#25221] [Q&A Meshing]
I am trying to pre-split a large mesh which involves LowerDBlockFromSidesetGenerator, however the process always segfaults whenever it begins writing files. An MWE command and input file are given at the bottom, along with the error message. Some observations:

- Splitting works fine with other mixed-dimensional meshes (e.g. combined 3D cube and 2D rectangle).
- Some of the files appear to write without issue before the segfault. My guess is these are sections of the mesh that don't contain the lower dimensional block.
- I have tried "pre-generating" the mesh using `--mesh-only`, and splitting a `.e` file instead, but the issue persists.
- This happens no matter how many pieces I split the mesh into, or how many processes/how much memory per process I allocate.

Any advice on how to work around this, or if there is likely to be a fix any time soon, would be much appreciated - thanks!

```bash
mpirun myapp-opt -i mesh.i --split-mesh 24 --split-file split.cpr
```

<details>
<summary>mesh.i</summary>

```
[Mesh]
  [cube]
    type = GeneratedMeshGenerator
    dim = 3
    nx = 10
    ny = 10
    nz = 10
  []
  [top]
    type = LowerDBlockFromSidesetGenerator
    input = cube
    sidesets = top
  []
[]
```
</details>

<details>
<summary>Error message</summary>

```
Splitting 24 ways...
    - writing 1 files per process...
Meta data are written into ./split.cpr/meta_data_mesh.rd.
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun noticed that process rank xx with PID xxxx on node xxxx exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------
```
</details>

**Comment by GiudGiud** [comment]
Thanks for providing the MWE and letting us know.
We'll debug this when possible

**Reply by GiudGiud** [reply]
In the mean time, you ll have to use replicated meshes and use less processors to be able to fit the problem in memory

**Reply by lindsayad** [reply]
@GiudGiud has opened https://github.com/idaholab/moose/issues/25236 for this

**Reply by ln53** [reply]
Thanks for the response, will do
